### PR TITLE
Fixes #37521 - Purge complete pulp tasks

### PR DESCRIPTION
**Pull request by ianballou**
#### What are the changes introduced in this pull request?
Adds completed Pulp task purging to orphan cleanup. Also adds a setting for how long to wait before considering tasks purge-able. 

#### Considerations taken when implementing this change?
Pulp by default purges complete tasks, but could optionally purge ones that stopped due to errors, for example. I opted to not delete these to help track down bugs.

#### What are the testing steps for this pull request?
Make some complete tasks in Pulp and try purging them by playing with the new setting.

### CODE REVIEW

**Comment by sjha4** [comment]
Code looks clean. Should we add a VCR test to catch any upgrade issues with the pulp API in the future?

**Comment by ianballou** [comment]
@sjha4 I added a test.

File: test/actions/pulp3/orchestration/remove_orphans_test.rb [hunk 1]
```diff
@@ -39,6 +39,8 @@ def setup
 
       sync_and_reload_repo(@repo, @primary)
 
+      Setting[:completed_pulp_task_protection_days] = 0
+      DateTime.expects(:now).returns(DateTime.new(3000,1,1))
```
Comment (sjha4) [inline test/actions/pulp3/orchestration/remove_orphans_test.rb:43]:
```suggestion
      DateTime.expects(:now).returns(DateTime.new(3000, 1, 1))
```
Rubocop unhappy..Code looks good..Should be good to go if tests are happy..

Comment (ianballou) [reply test/actions/pulp3/orchestration/remove_orphans_test.rb:43]: How rude of rubocop, I'll fix it.

**Comment by sjha4** [comment]
Tests probably failing because of the tests that call orphan cleanup to clean up test data for the test. We reduced this for a lot of tests but guess there's more.

**Comment by ianballou** [comment]
Looks like I just missed one VCR.

**Review by sjha4 — APPROVED** [review]
Ack.. 👍🏼

**Pull request merged**